Entities/HDFirearm.cpp

```diff
@@ -150,91 +144,79 @@ int HDFirearm::Create(const HDFirearm &reference)
 //                  is called. If the property isn't recognized by any of the base classes,
 //                  false is returned, and the reader's position is untouched.
 
-int HDFirearm::ReadProperty(std::string propName, Reader &reader)
-{
-    if (propName == "Magazine")
-    {
-        const Entity *pObj = g_PresetMan.GetEntityPreset(reader);
-        if (pObj)
-        {
-            m_pMagazineReference = dynamic_cast<const Magazine *>(pObj);
-
-            delete m_pMagazine;
-            m_pMagazine = dynamic_cast<Magazine *>(m_pMagazineReference->Clone());
+int HDFirearm::ReadProperty(std::string propName, Reader &reader) {
+    if (propName == "Magazine") {
+        RemoveAttachable(m_pMagazine);
+        const Entity *magazineEntity = g_PresetMan.GetEntityPreset(reader);
+        if (magazineEntity) {
+            m_pMagazineReference = dynamic_cast<const Magazine *>(magazineEntity);
+            m_pMagazine = dynamic_cast<Magazine *>(magazineEntity->Clone());
+            AddAttachable(m_pMagazine);
+            m_pMagazine->SetCollidesWithTerrainWhileAttached(false);
```

Comment:
It's fine as-is, yeah, because m_pMagazine is being set appropriately, and that's the most important part. These things filled in from read property exist solely in PresetMan and don't have to be perfect, really. Clone Create (i.e. `Create(reference)`) is called when you clone an object from PresetMan, and all that requires is that the hardcoded attachable pointer (in this case m_pMagazine) is set.

All that said, it might be cleaner to read into a new object and SetMagazine that object, but I think there might be lifetime problems, since that new object isn't a member variable. I could read into m_pMagazine as I currently do and call SetMagazine on that, but then I'd have to write it custom for each hardcoded attachable, and that's a bunch of pain in the ass work for no real benefit haha.